src/compute/src/extensions/arrange.rs

```diff
@@ -373,3 +373,83 @@ where
         })
     }
 }
+
+impl<G, V, T, R> ArrangementSize for Arranged<G, RowValAgent<V, T, R>>
+where
+    G: Scope<Timestamp = T>,
+    G::Timestamp: Lattice + Ord + Columnation,
+    V: Data + Columnation,
+    T: Lattice + Timestamp,
+    R: Semigroup + Columnation,
+{
+    fn log_arrangement_size(self) -> Self {
+        log_arrangement_size_inner(self, |trace| {
+            let (mut size, mut capacity, mut allocations) = (0, 0, 0);
+            let mut callback = |siz, cap| {
+                size += siz;
+                capacity += cap;
+                allocations += usize::from(cap > 0);
+            };
+            trace.map_batches(|batch| {
+                // batch.storage.keys.heap_size(&mut callback);
```

Comment:
@antiguru I commented these out, but probably there is no good reason. I'm just not sure what is needed for the associated trait. The `DatumContainer` is just a wrapper around a conventional container though, and it should have all the constraints needed to implement the trait (e.g. an inner container that implements the region stuff).

Reply:
This will allow you to use `heap_size`:

```diff
--- a/src/compute/src/row_spine.rs
+++ b/src/compute/src/row_spine.rs
@@ -105,6 +105,14 @@ mod container {
         staging: Vec<u8>,
     }

+    impl DatumContainer {
+        /// Estimate the memory capacity in bytes.
+        #[inline]
+        pub fn heap_size(&self, callback: impl FnMut(usize, usize)) {
+            self.inner.heap_size(callback);
+        }
+    }
+
     impl BatchContainer for DatumContainer {
         type PushItem = Row;
         type ReadItem<'a> = DatumSeq<'a>;
```

Reply:
This has changed, and it should be easy to update once I learn what the two `usize` arguments are. Probably "number and size of allocations", or the other way around, but I figure I'll ask / wait. :D

Reply:
The parameters are `(size, capacity)` in bytes for each allocated thingy (or an approximation thereof.)